Conclusion (#122)
chhoumann/p7: report/sections/Chapter 12 - Conclusion/Conclusion.tex

> The main purpose of this project was to create an internet-based application that would solve a problem. The developed system also needed to be tested. 

Reviewer: ```suggestion
The main goal of this project was to create an internet-based application that would solve some problem.
Furthermore, another requirement in the study regulation is that the developed system also needs to be tested for scalability. 
```

> In this project, we developed a system for lecturers of the \textit{Programming Paradigms} course, That would help them conduct the course. This system allowed them to define exercises with accompanying tests that students can solve. 

Reviewer: ```suggestion
In this project, we developed a system for lecturers of the \textit{Programming Paradigms} course that would help them conduct the course. This system allowed them to define exercises with accompanying tests that students can solve. 
```

> The students can then receive fast feedback without the need for a teaching assistant to verify if their solution is correct. 

Reviewer: ```suggestion```

> The students can then receive fast feedback without the need for a teaching assistant to verify if their solution is correct. 

Reviewer: ```suggestion
```

To create this application, we had to develop a working UI that the students and lecturers could use in order to interact with the system. The UI was developed to be user-friendly and provide a minimal amount of hindrance to solving exercises.
A backend was developed to serve the frontend to the users, and answer requests from the frontend. 

> The requests would then be rerouted to the relevant components of the system. 

Reviewer: Please rewrite this and give more details about the Test Runner, how we use the GHC interpreter with Haskell on the backend. Mention the queue system (probably not sweeping).
Generally, more details. This is far too vague.

Reviewer 2: Also tie it together with the suggestion above.

In order to store information regarding users and exercises, a database was implemented.
To be able to give the users feedback on their submitted exercises, we had to implement a component that would run the accompanying tests for an exercise on the submitted program. 
After the test runner has run the tests, it will send back the result to the user.

In order to ensure that the system would be scalable, we performed a stress test on different versions of the program. 
The stress tests showed that ... \todo{Insert stress test result}
To confirm that the UI had high usability, we performed a usability test.

> The test was performed on students who at the time were taking the \textit{Programming Paradigms}\todo{add textit to Programming Paradigmes in whole report} course.

Reviewer: ```suggestion
The test was performed on students who, at the time, were taking the \textit{Programming Paradigms}\todo{add textit to Programming Paradigmes in whole report} course.
```

> Based on the test the usability of the system was rated to be high.  

Reviewer: ```suggestion
The result of this test showed that the system usability was satisfying and usable in practice by students participating in the course.
```

> In this project we set out to create a scalable system, that would be an improvement to the \textit{Programming Paradigms} course.

Reviewer: Why are we saying this *again*? This literally repeats the very first sentence in the chapter.

> Our application can deliver swift evaluation of students' exercise submissions. Based on our tests the application will be scalable up to a whole class of students.

Reviewer: Needs to be rewritten - final sentence is outright wrong, and this does not summarize the points made in the conclusion.
